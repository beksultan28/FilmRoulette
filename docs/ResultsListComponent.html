<!DOCTYPE html>
<html lang="en">
<head>
    <meta charset="utf-8">
    <title>JSDoc: Class: ResultsListComponent</title>

    <script src="scripts/prettify/prettify.js"> </script>
    <script src="scripts/prettify/lang-css.js"> </script>
    <!--[if lt IE 9]>
      <script src="//html5shiv.googlecode.com/svn/trunk/html5.js"></script>
    <![endif]-->
    <link type="text/css" rel="stylesheet" href="styles/prettify-tomorrow.css">
    <link type="text/css" rel="stylesheet" href="styles/jsdoc-default.css">
</head>

<body>

<div id="main">

    <h1 class="page-title">Class: ResultsListComponent</h1>

    




<section>

<header>
    
        <h2><span class="attribs"><span class="type-signature"></span></span>ResultsListComponent<span class="signature">()</span><span class="type-signature"></span></h2>
        
    
</header>

<article>
    <div class="container-overview">
    
        

    

    
    <h4 class="name" id="ResultsListComponent"><span class="type-signature"></span>new ResultsListComponent<span class="signature">()</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    Search results componentDisplays search results and ResultsSortComponent
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_results-list_index.jsx.html">components/results-list/index.jsx</a>, <a href="components_results-list_index.jsx.html#line9">line 9</a>
    </li></ul></dd>
    

    

    

    
</dl>


















    
    </div>

    
        <h3 class="subsection-title">Extends</h3>

        


    <ul>
        <li>Component</li>
    </ul>


    

    

    

    

    

    

    

    
        <h3 class="subsection-title">Methods</h3>

        
            

    

    
    <h4 class="name" id=".getSortTitle"><span class="type-signature">(static) </span>getSortTitle<span class="signature">()</span><span class="type-signature"> &rarr; {String}</span></h4>
    

    



<div class="description">
    Generates title for ResultsSortComponent
</div>













<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_results-list_index.jsx.html">components/results-list/index.jsx</a>, <a href="components_results-list_index.jsx.html#line30">line 30</a>
    </li></ul></dd>
    

    

    

    
</dl>













<h5>Returns:</h5>

        
<div class="param-desc">
    title
</div>



<dl>
    <dt>
        Type
    </dt>
    <dd>
        
<span class="param-type">String</span>


    </dd>
</dl>

    





        
            

    

    
    <h4 class="name" id=".onSort"><span class="type-signature">(static) </span>onSort<span class="signature">(sortBy)</span><span class="type-signature"></span></h4>
    

    



<div class="description">
    onSort event handlerUpdates sortBy state with given dataCalls updateSortBy property functionPassed down to ResultsSortComponent child
</div>









    <h5>Parameters:</h5>
    

<table class="params">
    <thead>
    <tr>
        
        <th>Name</th>
        

        <th>Type</th>

        

        

        <th class="last">Description</th>
    </tr>
    </thead>

    <tbody>
    

        <tr>
            
                <td class="name"><code>sortBy</code></td>
            

            <td class="type">
            
                
<span class="param-type">String</span>


            
            </td>

            

            

            <td class="description last"></td>
        </tr>

    
    </tbody>
</table>






<dl class="details">

    

    

    

    

    

    

    

    

    

    

    

    

    
    <dt class="tag-source">Source:</dt>
    <dd class="tag-source"><ul class="dummy"><li>
        <a href="components_results-list_index.jsx.html">components/results-list/index.jsx</a>, <a href="components_results-list_index.jsx.html#line46">line 46</a>
    </li></ul></dd>
    

    

    

    
</dl>


















        
    

    

    
</article>

</section>




</div>

<nav>
    <h2><a href="index.html">Home</a></h2><h3>Classes</h3><ul><li><a href="DetailComponent.html">DetailComponent</a></li><li><a href="ResultsListComponent.html">ResultsListComponent</a></li><li><a href="ResultsSortComponent.html">ResultsSortComponent</a></li><li><a href="SearchFormComponent.html">SearchFormComponent</a></li></ul><h3>Global</h3><ul><li><a href="global.html#apiKey">apiKey</a></li><li><a href="global.html#detailItem">detailItem</a></li><li><a href="global.html#expectedResult">expectedResult</a></li><li><a href="global.html#expectedResults">expectedResults</a></li><li><a href="global.html#FooterComponent">FooterComponent</a></li><li><a href="global.html#getFilmApi">getFilmApi</a></li><li><a href="global.html#getFilmById">getFilmById</a></li><li><a href="global.html#getRecommendations">getRecommendations</a></li><li><a href="global.html#getRecommendationsApi">getRecommendationsApi</a></li><li><a href="global.html#recieveFilmById">recieveFilmById</a></li><li><a href="global.html#recieveFilms">recieveFilms</a></li><li><a href="global.html#ResultsItemComponent">ResultsItemComponent</a></li><li><a href="global.html#searchApi">searchApi</a></li><li><a href="global.html#searchBy">searchBy</a></li><li><a href="global.html#searchFilms">searchFilms</a></li><li><a href="global.html#searchResults">searchResults</a></li><li><a href="global.html#sortBy">sortBy</a></li><li><a href="global.html#sortFilms">sortFilms</a></li><li><a href="global.html#updateQuery">updateQuery</a></li><li><a href="global.html#updateSearchBy">updateSearchBy</a></li><li><a href="global.html#updateSortBy">updateSortBy</a></li></ul>
</nav>

<br class="clear">

<footer>
    Documentation generated by <a href="https://github.com/jsdoc3/jsdoc">JSDoc 3.5.5</a> on Sat Oct 28 2017 22:06:51 GMT+0600 (Central Asia Standard Time)
</footer>

<script> prettyPrint(); </script>
<script src="scripts/linenumber.js"> </script>
</body>
</html>