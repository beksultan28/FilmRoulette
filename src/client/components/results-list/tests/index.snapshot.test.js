import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';

import ResultsListComponent from '../index.jsx';

const updateSortBy = jest.fn();
const sortFilms = jest.fn();
const props = {
  title: 'count',
  searchResults: [{
    id: 1,
    title: 'Test movie',
    vote_average: 5,
    release_date: '2017-01-01',
  }],
  sortBy: 'movie',
  updateSortBy,
  sortFilms,
};

const renderer = new ShallowRenderer();
renderer.render(
  <ResultsListComponent
    {...props}
  />,
);
const resultsList = renderer.getRenderOutput();

test('ResultsListComponent renders correctly', () => {
  expect(resultsList).toMatchSnapshot();
});
