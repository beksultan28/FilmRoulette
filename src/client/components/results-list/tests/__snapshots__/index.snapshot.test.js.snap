// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ResultsListComponent renders correctly 1`] = `
<div>
  <ResultsSortComponent
    onSort={[Function]}
    sortBy="movie"
    title="1 items found"
  />
  <div
    className="container"
  >
    <div
      className="results-list"
    >
      <div
        className="row"
      >
        <div
          className="col-md-4 col-sm-6 col-xs-12"
        >
          <ResultsItemComponent
            item={
              Object {
                "id": 1,
                "release_date": "2017-01-01",
                "title": "Test movie",
                "vote_average": 5,
              }
            }
            searchBy={undefined}
          />
        </div>
      </div>
    </div>
  </div>
</div>
`;
